backend/src/packages/chaiNNer_ncnn/settings.py

```diff
@@ -53,12 +54,35 @@
         )
     )
 
+    package.add_setting(
+        NumberSetting(
+            label="Thread Count",
+            key="threads",
+            description="Number of threads for NCNN. Only affects CPU mode.",
+            default=ncnn.Net().opt.num_threads,
+            min=1,
+            max=ncnn.Net().opt.num_threads,
```

Comment:
Could these values be cached or promoted to constants? I don't know how expensive `ncnn.Net()`, so maybe it's not necessary. It would be nice to avoid the code duplication though.

Reply:
Good point. Yeah I can change that, hopefully by tomorrow.

Reply:
Done. I currently cache these settings regardless of whether upstream ncnn or chaiNNer's Vulkan fork is used; I don't think that will cause any problems (the settings exist in the Vulkan fork, and we don't actually change them when using the Vulkan fork), but if I've missed something and this causes breakage when using the Vulkan fork, let me know.

Also switched the Winograd/SGEMM settings to use the same approach, since there's no reason to duplicate the default settings in chaiNNer's codebase.